docs/user-guide/a1-06-spirv-target-specifics.md

```diff
@@ -0,0 +1,316 @@
+
+### [[vk::index(Y)]]
+Same as `index` layout qualifier in Vulkan. It is valid only when used with [[location(X)]]. For fragment shader outputs, the location and index specify the color output number and index receiving the values of the output. For outputs of all other shader stages, the location specifies a vector number that can be used to match against inputs in a subsequent shader stage.
+
+### [[vk::input_attachment_index(i)]]
+Same as `input_attachment_index` layout qualifier in Vulkan. It selects which subpass input is being read from. It is valid only when used on subpassInput type uniform variables.
+
+### [[vk::push_constant]]
+Same as `push_constant` layout qualifier in Vulkan. It is applicable only to a uniform block and it will be copied to a special memory location where GPU may have a more direct access to.
+
+### [vk::image_format(format : String)]
+Same as `[[vk::image_format("XX")]]` layout qualifier in DXC. Vulkan/GLSL allows the format string to be specified without the keyword, `image_format`.  Consider the following Slang code, as an example,
+```csharp
+[vk::image_format("r32f")] RWTexture2D<float> typicalTexture;
+```
+It will generate the following GLSL or SPIR-V code.
+> layout(r32f) uniform image2D typicalTexture_0;
+> %18 = OpTypeImage %float 2D 2 0 0 2 R32f
+
+### [vk::shader_record]
+Same as `shaderRecordEXT` layout qualifier in [GL_EXT_ray_tracing extension](https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_ray_tracing.txt).
+It can be used on a buffer block that represents a buffer within a shader record as defined in the Ray Tracing API.
+
+### [vk::aliased_pointer]
+When two memory pointers point to a same memory location, they are considered `aliased`. Compiler needs to be informed of such cases in order to generate the code that produces a consistant outcome. Slang can figure out whether a pointer needs to be treated as aliased or not, but you can explicitly mark a variable as aliased.
+
+### [vk::restrict_pointer]
+Opposite to [vk::aliased_pointer] in a way that it tells the compiler to treat the variable as not-aliased.
+
+### [vk::spirv_instruction(op : int, set : String = "")]
+When applied to a function, the function will use the `op` value corresponding to a SPIR-V instruction. You can also specify which instruction set you want to use as a second argument. E.G., `[[vk::spirv_instruction(1, "NonSemantic.DebugBreak")]]`
+
+
+Multiple entry points support
+-----------------------------
+
+Slang supports mutiple entry points when targeting SPIR-V.
+It is same to HLSL and SPIR-V that allows a shader source to have multiple entry points.
+Note that GLSL requires the entry point to be named, "main", and a shader source can have only one entry point.
```

Comment:
Mention that because of this, the current default behavior is to rename all entrypoints to "main" when emitting SPIRV. This default setting will prevent a SPIRV to have more than one entrypoint defined. To generate valid spirv with more than one entrypoints, use the `-fvk-use-entrypoint-name` compiler option to disable this renaming behavior and preserve the entry point names.

Reply:
Aha. I didn't know that the option exists for this reason.